stl/inc/flat_set

```diff
@@ -25,16 +25,21 @@ _STL_DISABLE_CLANG_WARNINGS
 
 _STD_BEGIN
 
-template <class _Ty>
-struct _NODISCARD _Clear_scope_guard {
-    _Ty* _Clearable;
-    ~_Clear_scope_guard() {
-        if (_Clearable) {
-            _Clearable->clear();
+template <class _Ty, bool _Noexcept = false>
+struct _NODISCARD _Clear_guard {
+    _Ty* _Target;

```

Comment:
renamed to match with existing practices
about `_Noexcept`:
1. It has some optimization effect. See [this link](https://godbolt.org/#g:!((g:!((g:!((h:codeEditor,i:(filename:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,selection:(endColumn:2,endLineNumber:27,positionColumn:2,positionLineNumber:27,selectionStartColumn:2,selectionStartLineNumber:27,startColumn:2,startLineNumber:27),source:'%23include+%3Ctype_traits%3E%0A%23include+%3Cvector%3E%0Atemplate+%3Cclass+_Ty,+bool+_Noexcept+%3D+false%3E%0Astruct+_NODISCARD+_Clear_guard+%7B%0A++++_Ty*+_Target%3B%0A++++~_Clear_guard()+%7B%0A++++++++if+(_Target)+%5B%5Bunlikely%5D%5D+%7B%0A++++++++++++_Target-%3Eclear()%3B%0A++++++++%7D%0A++++%7D%0A%7D%3B%0A%0Atemplate+%3Cclass+_Ty%3E%0Astruct+%5B%5Bmaybe_unused%5D%5D+_NODISCARD+_Clear_guard%3C_Ty,+true%3E+%7B%0A++++_Ty*+_Target%3B+//+no+effect%0A%7D%3B%0A%0Aauto+test()%7B%0A++++std::vector%3Cint%3E+a(10,+!'0!')%3B%0A%0A++++_Clear_guard%3Cstd::vector%3Cint%3E/*,true*/%3E+_Cg%7B%26a%7D%3B%0A++++std::vector%3Cint%3E+b%3Dstd::move(a)%3B+//+should+be+noexcept.%0A++++_Cg._Target%3Dnullptr%3B+//+always+goes+here.%0A++++//+however+still+results+in+larger+code+than+%22true%22+specialization.%0A++++//+this+suggests+the+if+(_Target)+branch+cannot+be+elided.%0A++++return+b%3B%0A%7D'),l:'5',n:'0',o:'C%2B%2B+source+%231',t:'0')),k:50,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((h:compiler,i:(compiler:vcpp_v19_37_x64,deviceViewOpen:'1',filters:(b:'0',binary:'1',binaryObject:'1',commentOnly:'0',debugCalls:'1',demangle:'0',directives:'0',execute:'1',intel:'0',libraryCode:'0',trim:'1'),flagsViewOpen:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,libs:!(),options:'/std:c%2B%2B20+/O2',overrides:!(),selection:(endColumn:1,endLineNumber:1,positionColumn:1,positionLineNumber:1,selectionStartColumn:1,selectionStartLineNumber:1,startColumn:1,startLineNumber:1),source:1),l:'5',n:'0',o:'+x64+msvc+v19.37+(Editor+%231)',t:'0')),k:50,l:'4',n:'0',o:'',s:0,t:'0')),l:'2',n:'0',o:'',t:'0')),version:4).
2. The name might not be accurate; `_Known_noexcept` might be more accurate.
3. It doesn't take strong-guarantee cases into consideration, as I find it too hard to do decisions based on this.

Reply:
temporarily removed in the new pr.